### PR TITLE
Delete `TxGovernanceActions`

### PR DESCRIPTION
**Pull request by newhoggy**
# Changelog

```yaml
- description: |
    Changes:
    * Deleted `TxGovernanceActions`
    * New generators: `genProposals`, `genProposal`, `genVotingProcedures`
    * New test constraints functions: `shelleyBasedEraTestConstraints`, `shelleyToBabbageEraTestConstraints`, `conwayEraOnwardsTestConstraints`
    * New era functions: `inEraFeatureMaybe`, `inShelleyBasedEraFeatureMaybe`, `maybeFeatureInShelleyBasedEra`
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

Delete `TxGovernanceActions`

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] The change log section in the PR description has been filled in
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - round trip tests
  - integration tests
  See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] The changelog section in the PR is updated to describe the change
- [ ] Self-reviewed the diff

### CODE REVIEW

File: cardano-api/gen/Test/Gen/Cardano/Api/Typed.hs [hunk 1]
```diff
@@ -1110,17 +1113,15 @@ genGovernancePollAnswer =
...
-genTxGovernanceActions era = fromMaybe (pure TxGovernanceActionsNone) $ do
-  sbe <- join $ requireShelleyBasedEra era
-  supported <- featureInShelleyBasedEra Nothing Just sbe
-  let proposals = Gen.list (Range.constant 0 10) $ genProposal supported
-  pure $ TxGovernanceActions supported <$> proposals
-  where
-    genProposal :: ConwayEraOnwards era
-                -> Gen (Proposal era)
-    genProposal = \case
-      ConwayEraOnwardsConway -> fmap Proposal Q.arbitrary
```
Comment (newhoggy) [inline cardano-api/gen/Test/Gen/Cardano/Api/Typed.hs:1123]: Generators are much simpler now.

File: cardano-api/gen/Test/Gen/Cardano/Api/Typed.hs [hunk 2]
```diff
@@ -1110,17 +1113,15 @@ genGovernancePollAnswer =
...
+genProposals :: forall era. ConwayEraOnwards era -> Gen [Proposal era]
+genProposals w =
+  conwayEraOnwardsTestConstraints w
+    $ Gen.list (Range.constant 1 10)
+    $ genProposal w
```
Comment (newhoggy) [inline cardano-api/gen/Test/Gen/Cardano/Api/Typed.hs:1120]: Generators are much simpler now.

File: cardano-api/internal/Cardano/Api/Governance/Actions/ProposalProcedure.hs [hunk 3]
```diff
@@ -43,21 +41,6 @@ import           Data.Maybe (fromMaybe)
...
-data TxGovernanceActions era where
-  TxGovernanceActionsNone :: TxGovernanceActions era
-
-  TxGovernanceActions
-    :: ConwayEraOnwards era
-    -> [Proposal era]
-    -> TxGovernanceActions era
-
-deriving instance IsShelleyBasedEra era => Show (TxGovernanceActions era)
-deriving instance IsShelleyBasedEra era => Eq (TxGovernanceActions era)
```
Comment (newhoggy) [inline cardano-api/internal/Cardano/Api/Governance/Actions/ProposalProcedure.hs:59]: Delete this type.

File: cardano-api/internal/Cardano/Api/TxBody.hs [hunk 4]
```diff
@@ -2702,62 +2701,51 @@ fromLedgerTxBody
...
-        $ toList
-        $ body_ ^. L.proposalProceduresTxBodyL
+  -> Maybe (Featured ConwayEraOnwards era [Proposal era])
+fromLedgerProposalProcedures sbe body =
+  inShelleyBasedEraFeatureMaybe sbe $ \w ->
+    conwayEraOnwardsConstraints w
+      $ Featured w
+      $ fmap Proposal
+      $ toList
+      $ body ^. L.proposalProceduresTxBodyL
```
Comment (newhoggy) [inline cardano-api/internal/Cardano/Api/TxBody.hs:2737]: Conversions are a bit simpler.

**Review by carbolymer — APPROVED** [review]

**Pull request merged**